src/screenshot/screenshot.c

```diff
@@ -35,17 +35,140 @@ static bool exec_screenshooter(const char *path) {
 
 	return stat == 0;
 }
+static bool exec_screenshooter_with_coordinates(const char *path,
+		char *coords) {
+	pid_t pid = fork();
+	if (pid < 0) {
+		perror("fork");
+		return false;
+	} else if (pid == 0) {
+		char cmd[strlen(coords) + strlen(path) + 14];
+		sprintf(cmd, "grim -g \"%s\" -- %s", coords, path);
+		execl("/bin/sh", "/bin/sh", "-c", cmd, NULL);
+		perror("execl");
+		exit(127);
+	}
+
+	int stat;
+	if (waitpid(pid, &stat, 0) < 0) {
+		perror("waitpid");
+		return false;
+	}
+
+	return stat == 0;
+}
+
+static pid_t spawn_chooser(int chooser_out[2]) {
+	pid_t pid = fork();
+	if (pid < 0) {
+		perror("fork");
+	} else if (pid == 0) {
+		close(chooser_out[0]);
+
+		dup2(chooser_out[1], STDOUT_FILENO);
+		close(chooser_out[1]);
+
+		execl("/bin/sh", "/bin/sh", "-c", "slurp", NULL);
+
+		perror("execl");
+		_exit(127);
+	}
+
+	close(chooser_out[1]);
+	return pid;
+}
+
+static char *exec_coordinates_selector() {
+	size_t size = 0;
+
+	int chooser_out[2];
+	if (pipe(chooser_out) == -1) {
+		perror("pipe chooser_out");
+		return false;
+	}
+
+	pid_t pid = spawn_chooser(chooser_out);
+	if (pid < 0) {
+		logprint(ERROR, "Failed to fork chooser");
+		close(chooser_out[0]);
+		return false;
+	}
+
+	FILE *f = fdopen(chooser_out[0], "r");
+	if (f == NULL) {
+		perror("fopen pipe chooser_out");
+		close(chooser_out[0]);
+		return false;
+	}
+
+	char *result;
+	ssize_t nread = getline(&result, &size, f);
+	fclose(f);
+	close(chooser_out[0]);
+	if (nread < 0) {
+		perror("getline failed");
+		return false;
+	}
+	char *coordinates = strtok(result, "\n");
+	if (coordinates == NULL) {
+		perror("Failed to get coordinates");
+		return false;
+	}
+	return coordinates;
+}
 
 static int method_screenshot(sd_bus_message *msg, void *data,
 		sd_bus_error *ret_error) {
 	int ret = 0;
 
+	bool interactive = false;
+
 	char *handle, *app_id, *parent_window;
 	ret = sd_bus_message_read(msg, "oss", &handle, &app_id, &parent_window);
 	if (ret < 0) {
 		return ret;
 	}
-	// TODO: read options
+
+	sd_bus_message_enter_container(msg, 'a', "{sv}");
+	if (ret < 0) {
+		return ret;
+	}
+	logprint(INFO, "dbus: handle: %s", handle);
+	logprint(INFO, "dbus: app_id: %s", app_id);
```

Comment:
INFO is for end-user messages. I think this should be TRACE or DEBUG, or removed altogether.